backend/substrapp/tests/tests_misc.py

```diff
@@ -99,3 +136,29 @@ def test_query_tuples(self):
         with patch('substrapp.ledger_utils.query_ledger') as mquery_ledger:
             mquery_ledger.return_value = None
             query_tuples('testtuple', 'data_owner')
+
+    def test_path_traversal(self):
+        # Zip
+        with patch('substrapp.utils.zipfile.is_zipfile') as mock_is_zipfile, \
+                patch('substrapp.utils.ZipFile') as mock_zipfile:
+            mock_is_zipfile.return_value = True
+            mock_zipfile.return_value = MockArchive()
+
+            self.assertRaises(Exception,
+                              uncompress_path('', DIRECTORY))
+
+        # Tar
+        with patch('substrapp.utils.zipfile.is_zipfile') as mock_is_zipfile, \
+                patch('substrapp.utils.tarfile.is_tarfile') as mock_is_tarfile, \
+                patch('substrapp.utils.tarfile.open') as mock_tarfile:
+            mock_is_zipfile.return_value = False
+            mock_is_tarfile.return_value = True
+            mock_tarfile.return_value = MockArchive()
+
+            self.assertRaises(Exception,
+                              uncompress_path('', DIRECTORY))
+
+        # Models
+        with self.assertRaises(Exception):
+            model_dst_path = os.path.join(DIRECTORY, f'model/../../hackermodel')
+            raise_if_path_traversal([model_dst_path], os.path.join(DIRECTORY, 'model/'))
```

Comment:
What about zipping and unzipping (same for tar) real archives instead of using mocking? 
It could be a little easier to maintain these tests and would allow to test a bit more?

Reply:
I thought about that ! But as we only care about filename path it easier to use a mock !

Reply:
I prefer to avoid mocking when I can. Seems ok to me.